datafusion/core/src/physical_plan/sorts/sort.rs

```diff
@@ -847,29 +853,54 @@ impl ExecutionPlan for SortExec {
 
         trace!("End SortExec's input.execute for partition: {}", partition);
 
-        let mut sorter = ExternalSorter::new(
-            partition,
-            input.schema(),
-            self.expr.clone(),
-            context.session_config().batch_size(),
-            self.fetch,
-            execution_options.sort_spill_reservation_bytes,
-            execution_options.sort_in_place_threshold_bytes,
-            &self.metrics_set,
-            context.runtime_env(),
-        );
+        if let Some(fetch) = self.fetch.as_ref() {
```

Comment:
In this prototype I simply used the `TopK` implementation for `SortExec` when there was a fetch as it produces the same output.

However, longer term I think we should make a real `TopKExec` `ExecutionPlan` so that the optimizers know about it and can avoid repartitioning / trying to do anything else fancy

Reply:
I wonder if there is some threshold at which the fetch is large enough that sorting is the better approach, certainly if the fetch is large enough that we need to spill

Reply:
Thank you @tustvold  and @gruuya  -- these are good ideas. 

Maybe we can figure out how to switch between sort and TopK dynamically -- like if TopK exceeds some memory threshold revert back to the normal ExternalSorter. Let me think about this 🤔